File: wildland-admin-manager/docker/Dockerfile.swift
FROM swift:5.6.1-focal

# Run with:
#
# docker-compose -f wildland-admin-manager/docker/docker-compose.yml run --rm wildland-sdk-swift

RUN mkdir -p wildland-core/wildland-admin-manager /root/.cargo
COPY . wildland-core
RUN apt-get -qy update && apt-get install -y curl g++
WORKDIR /wildland-core/wildland-admin-manager
# need to upgrade Rust toolchain to be able to compile the latest swift-bridge (trigerred by `cargo build`)
RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y \
    && printf '[registries]\nwl-dev = { index = "https://crates.wildland.dev/git/index" }\n' > /root/.cargo/config.toml \
    && mkdir -p wildland_swift \
    && . $HOME/.cargo/env \
    && cargo clean \
    && SWIFT_BRIDGE_OUT_DIR="$PWD/wildland_swift" cargo build --features "bindings,mocks" \
    && cp test/ffi/test.swift wildland_swift/main.swift \
    && swiftc -L ../target/debug -lwildland_admin_manager -lstdc++ \
    -I wildland_swift -import-objc-header \
    swift_header.h \
    ./wildland_swift/SwiftBridgeCore.swift \
    ./wildland_swift/wildland/wildland.swift \
    wildland_swift/main.swift \
    -o wildland_swift/swift_app

# Run the test. If everything was completed successfully, `echo $?` should return `0` exit code.

ENTRYPOINT ["./wildland_swift/swift_app"]

File: wildland-admin-manager/docker/Dockerfile.java
FROM wildland-sdk-base:latest

# Run with:
#
# docker-compose -f wildland-admin-manager/docker/docker-compose.yml run --rm wildland-sdk-java

ENV CC=g++
ENV JDK_INC_DIR=/usr/lib/jvm/java-17-openjdk-amd64/include

RUN apt-get -qy update && apt-get install -y swig openjdk-17-jdk

RUN mkdir -p wildland_java \
    && cd "${WLTMP}" \
    && swig -java -c++ -w516,503,476,302 -outdir ../wildland_java ../wildland.i \
    && mv ../wildland_wrap.cxx . \
    && ${CC} -fpermissive -shared -fPIC --std=c++14 -w \
    wildland_wrap.cxx ffi_cxx.rs.cc \
    -L../../target/debug \
    -lwildland_admin_manager \
    -I${JDK_INC_DIR} \
    -I${JDK_INC_DIR}/linux \
    -o ../wildland_java/libwildland.so

RUN cp test/ffi/test.java wildland_java/test.java \
    && cd wildland_java \
    && javac test.java

# Run the test. If everything was completed successfully, `echo $?` should return `0` exit code.

ENTRYPOINT ["java", "-cp", "wildland_java", "-Djava.library.path=./wildland_java", "main"]
